crates/control-interface/src/lib.rs

```diff
@@ -1,56 +1,96 @@
 //! # Control Interface Client
 //!
-//! This library provides a client API for consuming the wasmCloud control interface over a
-//! NATS connection. This library can be used by multiple types of tools, and is also used
-//! by the control interface capability provider and the wash CLI
+//! This library provides a client API for consuming the [wasmCloud control interface][docs-control-interface]
+//! over a NATS connection.
+//!
+//! This library can be used by multiple types of tools, and is also used
+//! by the control interface capability provider and the [`wash` CLI][wash-cli].
 //!
 //! ## Usage
-//! All of the [Client] functions are handled by a wasmCloud host running in the specified lattice.
+//!
+//! All of the [`Client`] functions are handled by a wasmCloud host running in the specified lattice.
+//!
 //! Each function returns a `Result<CtlResponse<T>>` wrapper around the actual response type. The outer
 //! result should be handled for protocol (timeouts, no hosts available) and deserialization errors (invalid response payload).
 //! The inner result is the actual response from the host(s) and should be handled for application-level errors.
+//!
+//! [docs-control-interface]: <https://wasmcloud.com/docs/hosts/lattice-protocols/control-interface>
+//! [wash-cli]: <https://wasmcloud.com/docs/ecosystem/wash>
 
 use serde::{Deserialize, Serialize};
 
 mod broker;
 mod otel;
 
 pub mod client;
-pub use client::{collect_sub_timeout, Client, ClientBuilder};
+pub use client::{Client, ClientBuilder};
 
 mod types;
 pub use types::component::*;
 pub use types::ctl::*;
 pub use types::host::*;
-pub use types::link::InterfaceLinkDefinition;
 pub use types::provider::*;
 pub use types::registry::*;
 pub use types::rpc::*;
 
-/// Identifier of one or more entities on the lattice used for addressing. May take many forms, such as:
-/// - component public key
-/// - provider public key
-/// - opaque string
-pub type LatticeTarget = String;
+pub use wasmcloud_core::{
+    ComponentId, InterfaceLinkDefinition, KnownConfigName, LatticeTarget, LinkName, WitInterface,
+    WitNamespace, WitPackage,
+};
+
+/// Enumeration of all kinds of identifiers in a wasmCloud lattice
+#[allow(unused)]
+#[non_exhaustive]
```

Comment:
Thoughts on dropping non_exhaustive on non public members?

Reply:
So... it turns out this isn't even used externally :) it's actually just used internally, and only to check if the identifiers are legit/usabble....

Reply:
Oh yeah, I just meant what do you think about removing `non_exhaustive` here since we're only constructing things internally

Reply:
Ahhhh -- yes, I can definitely do that!